cursor.go

```diff
@@ -233,11 +218,14 @@ func (c *Cursor) nextLocked(dest interface{}) (bool, error) {
 			} else {
 				c.buffer = append(c.buffer, value)
 			}
+			c.applyPendingSkips(true)
```

Comment:
exactly right. The issue is that after the buffer is refilled (from the atomic response) we need to apply the skips to the newly expanded records.